### DISCUSSION TITLE
Error when uploading multiple images

### DISCUSSION THREAD

**Discussion by kevinwaxi** [rahulhaque/laravel-filepond#33] [Q&A] [answered]
Hello there. 
Recently i got into this problem where when i try to upload multiple images.
Currently its a laravel inertia vue 3 (vite) project setup
And this is the error i got
`RahulHaque\Filepond\AbstractFilepond::decrypt(): Argument #1 ($data) must be of type string, array given, called in C:\Users\Kevin Wakhisi\Documents\Web Projects\clients\zerihun\habtam_ltd\vendor\rahulhaque\laravel-filepond\src\AbstractFilepond.php on line 51`
Below is the script
` 
          

            $product_cover_image = Filepond::field($request->cover)->getModel(); // Single Image
            $product_gallery_image = Filepond::field($request->gallery)->getModel(); // Multiple Images

            if ($product_cover_image) {
                try {
                    $product->addMediaFromDisk($product_cover_image->filepath, 'local')
                        ->toMediaCollection("products_cover");
                } catch (\Exception $e) {
                    dump($e);
                }

                Storage::disk('local')->delete($product_cover_image->filepath);
                $product_cover_image->forceDelete();
            }

            if ($product_gallery_image) {
                try {
                    foreach ($product_gallery_image as $gallery_image) {
                        $product->addMediaFromDisk($gallery_image->filepath, 'local')
                            ->toMediaCollection("product_gallery");
                    }
                } catch (\Exception $e) {
                    dump($e);
                }

                foreach ($product_gallery_image as $gallery_image) {
                    Storage::disk('local')->delete($gallery_image->filepath);
                    $gallery_image->forceDelete();
                }
            }
        });
`
Everything works apart from the multiple images

**Comment by rahulhaque** [comment, marked as answer]
@kevinwaxi what is the output of `$request->gallery` when you submit the form? I see you're using media library for storing files. [This](https://github.com/rahulhaque/laravel-filepond/issues/28#issuecomment-1328093675)  may also come in handy.

**Reply by kevinwaxi** [reply]
i actually did two things

1.  wrong never cleared some package that i tried earlier [https://github.com/Sopamo](url). so the config file (filepond.php) was not compatible
2. data from frontend side was not transformed as demonstrated.

I really appreciate your effort and reply
